core/src/utilities/helpers/iframe-helpers.js

```diff
@@ -104,3 +104,30 @@ export const getVisibleIframes = () => {
     .call(document.querySelectorAll('iframe'))
     .filter(item => item.style.display !== 'none');
 };
+
+export const getValidIframe = (event, window) => {
+  const iframes = document.querySelectorAll('.iframeContainer iframe');
+  const isOriginWindow = iframe =>
+    iframe &&
+    (event.source === iframe.contentWindow || event.source === window);
+  return Array.from(iframes).find(isOriginWindow);
+};
+
+export const urlMatchesTheDomain = (viewUrl = '', domain) => {
+  if (viewUrl.startsWith('/')) {
+    return domain === window.location.origin;
+  } else {
+    return viewUrl.startsWith(domain);
+  }
+};
+
+export const iframeIsSameDomain = (iframe, domain) => {
+  const sameDomain = urlMatchesTheDomain(iframe.luigi.viewUrl, domain);
+  iframe.luigi.trustedDomain = sameDomain ? domain : '';
+  return sameDomain;
+};
+
+export const sendPostMessage = (iframe, message) => {
```

Comment:
I would rename this to sth like 'sendMessageToIframe`, since `Post` is an implementation detail and this method is not valid if Luigi core sends a message to itself.